GetMagicCodeActivity/__tests__/handler.test.ts

```diff
@@ -1,30 +1,37 @@
-import { getActivityHandler } from "../handler";
+import { ActivityResultSuccess, getActivityHandler } from "../handler";
 import { context } from "../../__mocks__/durable-functions";
 import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
 import { aFiscalCode } from "../../__mocks__/mocks";
-import { TransientNotImplementedFailure } from "../../utils/durable";
+import * as E from "fp-ts/lib/Either";
+import { MagicLinkServiceClient } from "../utils";
 
 const aValidPayload = {
   family_name: "foo" as NonEmptyString,
   name: "foo" as NonEmptyString,
   fiscal_code: aFiscalCode
 };
 
-const mockMagicCodeServiceClient = {
-  getMagicCodeForUser: jest.fn().mockRejectedValue({ status: 501 })
-};
+const aValidMagicLink = "https://example.com/#token=abcde" as NonEmptyString;
+
+const mockMagicLinkServiceClient = ({
+  getMagicLinkToken: jest
+    .fn()
+    .mockResolvedValue(
+      E.right({ status: 200, value: { magic_link: aValidMagicLink } })
+    )
+} as unknown) as MagicLinkServiceClient;
 
 describe("GetMagicCodeActivity", () => {
-  it("should return a NOT_YET_IMPLEMENTED failure", async () => {
-    const result = await getActivityHandler(mockMagicCodeServiceClient)(
+  it("should return a success with a valid input", async () => {
+    const result = await getActivityHandler(mockMagicLinkServiceClient)(
       context as any,
       aValidPayload
     );
-    expect(TransientNotImplementedFailure.is(result)).toEqual(true);
+    expect(ActivityResultSuccess.is(result)).toEqual(true);
   });
 
   it("should return a FAILURE when the input is not valid", async () => {
-    const result = await getActivityHandler(mockMagicCodeServiceClient)(
+    const result = await getActivityHandler(mockMagicLinkServiceClient)(
       context as any,
       {}
```

Comment:
Include test cases for other failure scenarios

Reply:
added them in https://github.com/pagopa/io-functions-app/pull/285/commits/24e1bde8503879223b3d07a822bf1e8638544508